connector.py

```diff
@@ -106,10 +106,18 @@ def filter_json(self):
         # Listen for new messages from the chat service
         while True:
             try:
-                response = await self.connection.sync(
-                    self.connection.sync_token,
-                    timeout_ms=int(6 * 60 * 60 * 1e3),  # 6h in ms
-                    filter=self.filter_id)
+                try:
```

Comment:
This is not needed. The Sync is already in a try except block and it will retry the sync on any error. (the one below on send is not however so that's fine).